### PR TITLE
46 implement a structure to name and manage the picture setfolder

### PR DESCRIPTION
**Pull request by sylvanie85**
#46 
I made some fixes after i add the column name to picture_set table
I also add the `get_picture_sets_info` to retrieves directories information using the db
I made some fixes and tests for azure_storage_api functions
I add a trigger to set the picture_set name to the uuid if it's None

### CODE REVIEW

File: datastore/blob/azure_storage_api/__init__.py [hunk 1]
```diff
@@ -346,19 +346,30 @@ async def download_container(container_client, container_name, local_dir):
...
-    This function gets the list of blobs in a picture set folder
+    This function gets the names of blobs in a picture set folder
 
     Parameters:
     - container_client: the Azure container client
-    - tag: the tag to search for in the blobs ex: ""yourtagname"='firsttag' and "yourtagname2"='secondtag'"
+    - tag: the tag to search for in the blobs ex: 'folder_name'
 
-    Returns: the list of blobs
+    Returns: the names of blobs
```
Comment (Francois-Werbrouck) [inline datastore/blob/azure_storage_api/__init__.py:355]: It still is a list of the names

File: datastore/db/queries/picture/__init__.py [hunk 2]
```diff
@@ -17,42 +17,65 @@ class PictureSetNotFoundError(Exception):
...
+class GetPictureSetError(Exception):
+    pass
 
 """
 This module contains all the queries related to the Picture and PictureSet tables.
 """
 
-
-def new_picture_set(cursor, picture_set, user_id: str):
+def new_picture_set(cursor, picture_set, user_id: str, folder_name: str = None):
```
Comment (Francois-Werbrouck) [inline datastore/db/queries/picture/__init__.py:27]: I assume a picture_set still has a name on the tool even if it isn't named properly (ence why we make it optional). However I think we should still save a name if there is no name given to represent the default given name of the folder? WDYT?

File: datastore/blob/azure_storage_api/__init__.py [hunk 3]
```diff
@@ -112,24 +112,24 @@ async def get_blob(container_client, blob_name):
...
-            raise CreateDirectoryError(f"Folder:{folder_uuid} does not exist")
+        if not await is_a_folder(container_client, folder_name):
+            raise CreateDirectoryError(f"Folder:{folder_name} does not exist")
         else:
-            blob_name = "{}/{}.png".format(folder_uuid, image_uuid)
+            blob_name = "{}/{}.png".format(folder_name, image_uuid)
             metadata = {
                 "picture_uuid": f"{str(image_uuid)}",
-                "picture_set_uuid": f"{str(folder_uuid)}",
+                "picture_set_uuid": f"{str(folder_name)}",
```
Comment (Francois-Werbrouck) [inline datastore/blob/azure_storage_api/__init__.py:132]: It doesnt sit right with me to have a tag for their uuid and the set value to their name...
```suggestion
                "picture_set_uuid": f"{str(folder_uuid)}",
```

**Review by Francois-Werbrouck — CHANGES_REQUESTED** [review]
I see some work in db/queries/picture, but no new tests for test_db. Other than that, nice work implementing the naming !

File: datastore/db/bytebase/trigger_insert_picture_set_default_name.sql
Comment (Francois-Werbrouck) [inline datastore/db/bytebase/trigger_insert_picture_set_default_name.sql:1]: Great this fixes: 

> I assume a picture_set still has a name on the tool even if it isn't named properly (ence why we make it optional). However I think we should still save a name if there is no name given to represent the default given name of the folder? WDYT?

**Review by Francois-Werbrouck — APPROVED** [review]
LGTM! Nice work on the trigger!

File: datastore/blob/azure_storage_api/__init__.py [hunk 4]
```diff
@@ -270,7 +271,7 @@ async def get_image_count(container_client, folder_name):
             blob_list = container_client.list_blobs()
             count = 0
             for blob in blob_list:
-                if (blob.name.split("/")[0] == folder_uuid) and (
+                if (blob.name.split("/")[0] == folder_name) and (
                     blob.name.split(".")[-1] == "png"
                 ):
                     count += 1
```
Comment (ChromaticPanic) [inline datastore/blob/azure_storage_api/__init__.py:277]: How does this work with the different image formats? jpg, tif, tiff . do we save them all as png?

File: datastore/blob/azure_storage_api/__init__.py [hunk 5]
```diff
@@ -112,21 +112,22 @@ async def get_blob(container_client, blob_name):
...
-        if not await is_a_folder(container_client, folder_uuid):
-            raise CreateDirectoryError(f"Folder:{folder_uuid} does not exist")
+        if not await is_a_folder(container_client, folder_name):
+            raise CreateDirectoryError(f"Folder:{folder_name} does not exist")
         else:
-            blob_name = "{}/{}.png".format(folder_uuid, image_uuid)
+            blob_name = "{}/{}.png".format(folder_name, image_uuid)
             metadata = {
                 "picture_uuid": f"{str(image_uuid)}",
                 "picture_set_uuid": f"{str(folder_uuid)}",
```
Comment (ChromaticPanic) [inline datastore/blob/azure_storage_api/__init__.py:133]: How does this work with the different image formats? jpg, tif, tiff . do we save them all as png?

**Review by ChromaticPanic — APPROVED** [review]

**Pull request merged**